compiler/pre_hook_amd-llvm.cmake

```diff
@@ -49,10 +49,12 @@ set(LLVM_TARGETS_TO_BUILD "AMDGPU;X86" CACHE STRING "Enable LLVM Targets" FORCE)
 # Packaging.
 set(PACKAGE_VENDOR "AMD" CACHE STRING "Vendor" FORCE)
 
-# Build the device-libs as part of the core compiler so that clang works by
-# default (as opposed to other components that are *users* of the compiler).
+# Build device-libs and spirv-llvm-translator as part of the core
+# compiler default (as opposed to other components that are *users*
+# of the compiler).
 set(LLVM_EXTERNAL_ROCM_DEVICE_LIBS_SOURCE_DIR "${THEROCK_SOURCE_DIR}/compiler/amd-llvm/amd/device-libs")
-set(LLVM_EXTERNAL_PROJECTS "rocm-device-libs" CACHE STRING "Enable extra projects" FORCE)
+set(LLVM_EXTERNAL_SPIRV_LLVM_TRANSLATOR_SOURCE_DIR "${THEROCK_SOURCE_DIR}/compiler/spirv-llvm-translator")
+set(LLVM_EXTERNAL_PROJECTS "rocm-device-libs;spirv-llvm-translator" CACHE STRING "Enable extra projects" FORCE)
```

Comment:
How much does including spirv-llvm-translator affect binary size? Are there any other high level concerns/considerations we should be aware of about including it?